flash/data/callback.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, List, Sequence
+
+from pytorch_lightning.callbacks import Callback
+from pytorch_lightning.trainer.states import RunningStage
+from torch import Tensor
+
+
+class FlashCallback(Callback):
+
+    def on_load_sample(self, sample: Any, running_stage: RunningStage) -> None:
+        """Called once a sample has been loaded."""
+
+    def on_pre_tensor_transform(self, sample: Any, running_stage: RunningStage) -> None:
+        """Called once an object has been transformed."""
+
+    def on_to_tensor_transform(self, sample: Any, running_stage: RunningStage) -> None:
+        """Called once an object has been transformed to a tensor."""
+
+    def on_post_tensor_transform(self, sample: Tensor, running_stage: RunningStage) -> None:
+        """Called after `post_tensor_transform` """
+
+    def on_per_batch_transform(self, batch: Any, running_stage: RunningStage) -> None:
+        """Called after `per_batch_transform` """
+
+    def on_collate(self, batch: Sequence, running_stage: RunningStage) -> None:
+        """Called after `collate` """
+
+    def on_per_sample_transform_on_device(self, samples: Sequence, running_stage: RunningStage) -> None:
+        """Called after `per_sample_transform_on_device` """
+
+    def on_per_batch_transform_on_device(self, batch: Any, running_stage: RunningStage) -> None:
+        """Called after `per_batch_transform_on_device` """
+
+
+class ControlFlow(FlashCallback):
```

Comment:
This is a bit weird to me as we don't have such thing in Lightning. We just loop over all callbacks for each hook around the code. Do you prefer this object? Why?

Reply:
Yes, I prefer this object and I think we should add a similar one within Lightning. The callbacks logic should be fully managed by callbacks.